openpype/pipeline/farm/pyblish_functions.py

```diff
@@ -0,0 +1,871 @@
+    end = instance.data.get("frameEnd")
+    project_name = instance.context.data["project"]
+    anatomy = instance.context.data["anatomy"]  # type: Anatomy
+
+    # get latest version of subset
+    # this will stop if subset wasn't published yet
+
+    version = get_last_version_by_subset_name(
+        project_name,
+        instance.data.get("subset"),
+        asset_name=instance.data.get("asset")
+    )
+
+    # get its files based on extension
+    subset_resources = get_resources(
+        project_name, version, representation.get("ext")
+    )
+    r_col, _ = clique.assemble(subset_resources)
+
+    # if override remove all frames we are expecting to be rendered,
+    # so we'll copy only those missing from current render
+    if instance.data.get("overrideExistingFrame"):
+        for frame in range(start, end + 1):
+            if frame not in r_col.indexes:
+                continue
+            r_col.indexes.remove(frame)
+
+    # now we need to translate published names from representation
+    # back. This is tricky, right now we'll just use same naming
+    # and only switch frame numbers
+    resource_files = []
+    r_filename = os.path.basename(
+        representation.get("files")[0])  # first file
+    op = re.search(R_FRAME_NUMBER, r_filename)
+    pre = r_filename[:op.start("frame")]
+    post = r_filename[op.end("frame"):]
+    assert op is not None, "padding string wasn't found"
+    for frame in list(r_col):
+        fn = re.search(R_FRAME_NUMBER, frame)
+        # silencing linter as we need to compare to True, not to
+        # type
+        assert fn is not None, "padding string wasn't found"
+        # list of tuples (source, destination)
+        staging = representation.get("stagingDir")
+        staging = anatomy.fill_root(staging)
+        resource_files.append(
+            (frame, os.path.join(
+                staging, "{}{}{}".format(pre, fn["frame"], post)))
+        )
+
+    # test if destination dir exists and create it if not
+    output_dir = os.path.dirname(representation.get("files")[0])
+    if not os.path.isdir(output_dir):
+        os.makedirs(output_dir)
+
+    # copy files
+    for source in resource_files:
+        speedcopy.copy(source[0], source[1])
+        log.info("  > {}".format(source[1]))
+
+    log.info("Finished copying %i files" % len(resource_files))
+
+
+def attach_instances_to_subset(attach_to, instances):
+    """Attach instance to subset.
+
+    If we are attaching to other subsets, create copy of existing
+    instances, change data to match its subset and replace
+    existing instances with modified data.
+
+    Args:
+        attach_to (list): List of instances to attach to.
+        instances (list): List of instances to attach.
+
+    Returns:
+          list: List of attached instances.
+
+    """
+    #
+
+    new_instances = []
+    for attach_instance in attach_to:
+        for i in instances:
+            new_inst = copy.deepcopy(i)
```

Comment:
There recently have been issues with deep copying `instance.data` variables - so keep that in mind. #5171

Reply:
Didn't notice this would an issue during testing.

Reply:
Might be substance painter only but it somehow got a hickup there recently. Not sure what changed.